### PR TITLE
feat: add support for path params is useMutate

### LINKED ISSUE #255: Ability to provide path params to mutate method when using useMutate
Right now there is no way to provide path params when `mutate` is called.

```ts
/* generated code */
export const useUpdatePet = ({id, ...props}: UseUpdatePetProps) => 
  useMutate<Pet, Error, void, UpdatePetRequestRequestBody>("PATCH", `/pets/${id}`, props);

/* usage */
const { mutate } = useUpdatePet({ id: '' })

async function handleUpdate() {
  await mutate() // -> Right now we cannot pass id dynamically here
}
```
This causes issue when using it in with data-tables and grids.

I do not want to render/use hooks for each row individually. I would like to render/use it once in the parent component and pass the path params when I call mutate.

I have made a rough implementation for the same [here](https://github.com/vkbansal/restful-react/commit/7f8882b00711c2c3716ff69b49d66a4d8fccc059).

### PR DESCRIPTION
**Pull request by vkbansal**
# Why
Fixes #255

### CODE REVIEW

**Comment by vkbansal** [comment]
@fabien0102 can you take a look at this?

**Comment by fabien0102** [comment]
A bit busy today, but I keep the browser tab open to not forget 😁

**Comment by vkbansal** [comment]
Thanks @fabien0102

**Comment by vkbansal** [comment]
@fabien0102 Thanks for the review. I'll make the suggested changes.

File: src/scripts/tests/__snapshots__/import-open-api.test.ts.snap [hunk 1]
```diff
@@ -97,90 +97,98 @@ export type UseFindPetsProps = Omit<UseGetProps<Pet[], FindPetsQueryParams>, \\"
...
 /**
  * deletes a single pet based on the ID supplied
  */
-export const useDeletePet = (props: UseDeletePetProps) => useMutate<void, Error, void, number>(\\"DELETE\\", \`/pets\`, props);
+export const useDeletePet = (props: UseDeletePetProps) => useMutate<void, Error, void, number, void>(\\"DELETE\\", \`/pets\`, { ...props });
 
 
-export type UpdatePetProps = Omit<MutateProps<Pet, Error, void, UpdatePetRequestRequestBody>, \\"path\\" | \\"verb\\"> & {id: number};
+export interface UpdatePetPathParams {
+  id: number
```
[the remark this answers is not in the document]
Comment (vkbansal) [reply src/scripts/tests/__snapshots__/import-open-api.test.ts.snap:170]: Can you give an example?

File: src/Get.tsx [hunk 2]
```diff
@@ -78,6 +78,10 @@ export interface GetProps<TData, TError, TQueryParams> {
   children: (data: TData | null, states: States<TData, TError>, actions: Actions<TData>, meta: Meta) => React.ReactNode;
   /** Options passed into the fetch call. */
   requestOptions?: RestfulReactProviderProps["requestOptions"];
+  /**
+   * Query parameters
```
[the remark this answers is not in the document]
Comment (vkbansal) [reply src/Get.tsx]: Yes. Fixed it.

File: src/Mutate.tsx [hunk 3]
```diff
@@ -111,8 +115,8 @@ export interface MutateState<TData, TError> {
  * is a named class because it is useful in
  * debugging.
  */
-class ContextlessMutate<TData, TError, TQueryParams, TRequestBody> extends React.Component<
-  MutateProps<TData, TError, TQueryParams, TRequestBody> & InjectedProps,
+class ContextlessMutate<TData, TError, TQueryParams, TRequestBody, TPathParams = unknown> extends React.Component<
```
[the remark this answers is not in the document]
Comment (vkbansal) [reply src/Mutate.tsx]: Done

File: src/scripts/tests/__snapshots__/import-open-api.test.ts.snap [hunk 4]
```diff
@@ -97,90 +97,98 @@ export type UseFindPetsProps = Omit<UseGetProps<Pet[], FindPetsQueryParams>, \\"
  * Sed tempus felis lobortis leo pulvinar rutrum. Nam mattis velit nisl, eu condimentum ligula luctus nec. Phasellus semper velit eget aliquet faucibus. In a mattis elit. Phasellus vel urna viverra, condimentum lorem id, rhoncus nibh. Ut pellentesque posuere elementum. Sed a varius odio. Morbi rhoncus ligula libero, vel eleifend nunc tristique vitae. Fusce et sem dui. Aenean nec scelerisque tortor. Fusce malesuada accumsan magna vel tempus. Quisque mollis felis eu dolor tristique, sit amet auctor felis gravida. Sed libero lorem, molestie sed nisl in, accumsan tempor nisi. Fusce sollicitudin massa ut lacinia mattis. Sed vel eleifend lorem. Pellentesque vitae felis pretium, pulvinar elit eu, euismod sapien.
  * 
  */
-export const useFindPets = (props: UseFindPetsProps) => useGet<Pet[], Error, FindPetsQueryParams>(\`/pets\`, props);
+export const useFindPets = (props: UseFindPetsProps) => useGet<Pet[], Error, FindPetsQueryParams, void>(\`/pets\`, { ...props });
```
[the remark this answers is not in the document]
Comment (vkbansal) [reply src/scripts/tests/__snapshots__/import-open-api.test.ts.snap]: done

File: src/scripts/tests/__snapshots__/import-open-api.test.ts.snap (hunk 1 above)
[the remark this answers is not in the document]
Comment (vkbansal) [reply src/scripts/tests/__snapshots__/import-open-api.test.ts.snap:176]: Nevermind. Figured it out. Done.

**Comment by vkbansal** [comment]
@fabien0102 I've made the changes as suggested.

**Pull request merged**